File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> st;
        int i=0;
        int j=0;
        vector<int> res(temperatures.size(),0);
        for(int i=0;i<temperatures.size();i++)
        {
            while(!st.empty() && temperatures[st.top()]<temperatures[i])
            {
                int top=st.top();
                st.pop();
                res[top]=i-top;
            }
            st.push(i);
        }
        while(!st.empty())
        {
            res[st.top()]=0;
            st.pop();
        }
        return res;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack=[]
        operators=['*','/','+','-']
        for i in tokens:
            if i not in operators:
                stack.append(int(i))
            else:
                oper2=stack.pop()
                oper1=stack.pop()
                if i=='*':
                    res=oper2*oper1
                elif i=='/':
                    res=int(oper1/oper2)
                elif i=='-':
                    res=oper1-oper2
                else:
                    res=oper1+oper2
                stack.append(res)
        return stack[0]
        
            

        